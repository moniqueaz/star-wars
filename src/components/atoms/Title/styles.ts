import styled, { css, DefaultTheme } from 'styled-components';

const getColor = {
  light: (theme: DefaultTheme) => css`color: ${theme.colors.white}`,
  dark: (theme: DefaultTheme) => css`color: ${theme.colors.primary}`,
};

type WrapperProps = { color: 'light' | 'dark' }

export const Wrapper = styled.div<WrapperProps>`
  ${( { theme, color } ) => css`
    h1, h2, h3{
      ${getColor[color](theme)}
    }
  `}
`;

export const Title = styled.h1`
  ${( { theme } ) => css`
    font-size: ${theme.font.sizes.xxlarge};
  `}
`;

export const SessionTitle = styled.h2`
  ${( { theme } ) => css`
    font-size: ${theme.font.sizes.xlarge};
  `}
`;

export const SubTitle = styled.h3`
  ${( { theme } ) => css`
    font-size: ${theme.font.sizes.large};
  `}
`;
