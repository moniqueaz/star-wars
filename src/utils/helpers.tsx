export const fetcher = (url: string) => fetch(url).then(res => res.json());

export const getRandomInteger = (min: number, max: number) => {
  return Math.floor(Math.random() * (max - min)) + min;
};
