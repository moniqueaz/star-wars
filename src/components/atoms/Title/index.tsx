import * as S from './styles';

type TitleProps = {
  children?: React.ReactNode
  type?: 'title' | 'sessionTitle' | 'subTitle',
  color?: 'light' | 'dark'
}

const titleType = {
  title: (title: React.ReactNode) => <S.Title>{title}</S.Title>,
  sessionTitle: (title: React.ReactNode) => <S.SessionTitle>{title}</S.SessionTitle>,
  subTitle: (title: React.ReactNode) => <S.SubTitle>{title}</S.SubTitle>,
};

const Title = ( { type = 'title', children, color = 'light' }: TitleProps) => {
  if (!children) {
    return null;
  }

  return (
    <S.Wrapper color={color}>
      {titleType[type](children)}
    </S.Wrapper>
  );
};

export default Title;
