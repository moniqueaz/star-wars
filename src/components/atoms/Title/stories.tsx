import { Story, Meta } from '@storybook/react/types-6-0';
import Title from '.';

export default {
  title: 'Atoms/Title',
  component: Title,
} as Meta;

export const Default: Story = args => {
  return <Title {...args}>Titulo</Title>;
};

export const SessionTitle: Story = () => {
  return <Title type="sessionTitle">Titulo de SessÃ£o</Title>;
};

export const SubTitle: Story = () => {
  return <Title type="subTitle">Subtitulo</Title>;
};

export const Dark: Story = () => {
  return <Title color="dark">Title</Title>;
};

Dark.parameters = {
  backgrounds: {
    default: 'light',
  },
};
