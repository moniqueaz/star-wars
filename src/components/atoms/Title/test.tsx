import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import Title from '.';

describe('<Title />', () => {
  it('should render Title', () => {
    const { container } = renderWithTheme(<Title>Titulo</Title>);
    expect(screen.getByRole('heading', { name: /Titulo/i } )).toBeInTheDocument();
    expect(container.querySelector('h1')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  } );

  it('should render Title', () => {
    const { container } = renderWithTheme(<Title type="sessionTitle">Titulo de Sessão</Title>);
    expect(screen.getByRole('heading', { name: /Titulo de Sessão/i } )).toBeInTheDocument();
    expect(container.querySelector('h2')).toBeInTheDocument();
  } );

  it('should render Subtitle', () => {
    const { container } = renderWithTheme(<Title type="subTitle">Subtitulo</Title>);
    expect(screen.getByRole('heading', { name: /Subtitulo/i } )).toBeInTheDocument();
    expect(container.querySelector('h3')).toBeInTheDocument();
  } );

  it('should render without Title', () => {
    const { container } = renderWithTheme(<Title />);
    expect(container.querySelector('h1')).not.toBeInTheDocument();
  } );

  it('should render title dark', () => {
    const { container } = renderWithTheme(<Title color="dark">Titulo</Title>);
    expect(container.querySelector('h1')).toHaveStyle( { color: '#1A4173' } );
  } );

  it('should render title light', () => {
    const { container } = renderWithTheme(<Title>Titulo</Title>);
    expect(container.querySelector('h1')).toHaveStyle( { color: '#f1f1f1' } );
  } );
} );
