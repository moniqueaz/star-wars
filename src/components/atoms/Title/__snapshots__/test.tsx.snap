// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render Title 1`] = `
.c0 h1,
.c0 h2,
.c0 h3 {
  color: #f1f1f1;
}

.c1 {
  font-size: 2.8rem;
}

<div
  class="c0"
  color="light"
>
  <h1
    class="c1"
  >
    Titulo
  </h1>
</div>
`;
